internal/scan-manager/webapp/src/main/java/org/wso2/security/tools/scanmanager/webapp/service/ScanServiceImpl.java

```diff
@@ -0,0 +1,328 @@
+            if (!scanDirectoryLocation.exists() && !scanDirectoryLocation.mkdirs()) {
+                throw new ScanManagerWebappException("Error occurred while creating the scan directory");
+            }
+
+            // store the uploaded files in a temp scan directory.
+            for (Map.Entry<String, MultipartFile> entry : fileMap.entrySet()) {
+                String unifiedFileName = entry.getKey() + "." +
+                        FilenameUtils.getExtension(entry.getValue().getOriginalFilename());
+                Path unifiedFilePath = Paths.get(scanDirectoryLocation.toPath().toString(), unifiedFileName);
+                writeToFile(entry.getValue(), unifiedFilePath);
+                storedFileMap.put(entry.getKey(), unifiedFilePath.toString());
+            }
+
+            Iterator<Map.Entry<String, String>> iter = parameterMap.entrySet().iterator();
+            while (iter.hasNext()) {
+                Map.Entry<String, String> param = iter.next();
+                if (param.getValue().isEmpty()) {
+                    iter.remove();
+                } else {
+                    if (param.getKey().contains("@byURL")) {
+
+                        // if the param map contains files as a URL (param names containing "@byURL"), add them to
+                        // a separate map to download later.
+                        String fileExtension = FilenameUtils.getExtension(param.getValue());
+                        if (StringUtils.isBlank(fileExtension)) {
+                            throw new ScanManagerWebappException("Unable to identify the file extension from the " +
+                                    "file URL: " + param.getValue());
+                        }
+                        String fileName =
+                                StringUtils.substringBeforeLast(param.getKey(), "@") + "." + fileExtension;
+
+                        if (!filesToBeDownloadedFromURL.containsKey(fileName)) {
+                            filesToBeDownloadedFromURL.put(fileName, param.getValue());
+                        }
+                    }
+                }
+            }
+
+            // begin the pre scans tasks and initiate the scan submission.
+            new Thread(() -> beginScanSubmit(storedFileMap, parameterMap, filesToBeDownloadedFromURL, scanDirectory,
+                    scanWaiting), "BeginScanSubmitToScanManagerAPI").start();
+        } catch (ScanManagerWebappException e) {
+            logService.insertError(scanWaiting, e);
+
+            // Delete local scan artifacts directory.
+            try {
+                FileUtils.deleteDirectory(scanDirectoryLocation);
+            } catch (IOException ex) {
+                logService.insertError(scanWaiting, ex);
+            }
+        }
+        return scanWaiting;
+    }
+
+    private void writeToFile(MultipartFile file, Path filepath) throws ScanManagerWebappException {
```

Comment:
It's better to avoid acronyms like `os`, `fio`, `fos`, `is`, etc. Shall we call this `destinationOutputStream`? Code will be much more readable.